Sources/ComposedUI/CollectionView/UICollectionViewSection.swift

```diff
@@ -0,0 +1,45 @@
+import UIKit
+import Composed
+
+@available(*, deprecated, renamed: "SingleUICollectionViewSection")
+public typealias CollectionSectionProvider = SingleUICollectionViewSection
+
+/// A section in a `UICollectionView`.
+public protocol UICollectionViewSection: Section {
+    func collectionViewElementsProvider(with traitCollection: UITraitCollection) -> UICollectionViewSectionElementsProvider
+}
+
+public protocol SingleUICollectionViewSection: UICollectionViewSection {
+    func section(with traitCollection: UITraitCollection) -> CollectionSection
+}
+
+extension SingleUICollectionViewSection {
+    public func collectionViewElementsProvider(with traitCollection: UITraitCollection) -> UICollectionViewSectionElementsProvider {
+        return section(with: traitCollection)
+    }
+}
+
+@available(*, deprecated, renamed: "UICollectionViewSectionElementsProvider")
+public typealias CollectionElementsProvider = UICollectionViewSectionElementsProvider
+
+public protocol UICollectionViewSectionElementsProvider {
+    var header: CollectionSupplementaryElement? { get }
+    var footer: CollectionSupplementaryElement? { get }
+    var numberOfElements: Int { get }
+
+    func cell(for index: Int) -> CollectionCellElement
+}
+
+extension UICollectionViewSectionElementsProvider {
+    public var isEmpty: Bool { return numberOfElements == 0 }
```

Comment:
I wonder if this should be (to avoid issues with negative values):
```suggestion
    public var isEmpty: Bool { return numberOfElements <= 0 }
```

Reply:
Ditto for any other similar code.

Reply:
`numberOfElements` should never be negative, would an assertion be better?